renpy.el

```diff
@@ -1529,6 +1529,109 @@ Uses `renpy-beginning-of-block', `renpy-end-of-block'."
   (renpy-end-of-block)
   (exchange-point-and-mark))
 
+;;;; Completion
+
+(defcustom renpy-setup-completion t
+  "Non-nil means Renpy mode sets up completion for the buffer."
+  :type 'boolean
+  :group 'renpy)
+
+(defun renpy--completion-context ()
+  "Return the completion category for point."
+  ;; TODO: Make sure the point is not within a comment or a string.
+  ;; TODO: More specific contexts: show text, show expression, ...
+  (save-excursion
+    (skip-syntax-backward "w_")
+    (let* ((_ (skip-syntax-backward " "))
+           (prev  (thing-at-point 'symbol)))
+      (cond
+       ;; call <point>
+       ((equal prev "call") :label)
+       ;; jump <point>
+       ((equal prev "jump") :label)
+       ;; show/scene/hide <point>
+       ((member prev '("show" "scene" "hide")) :image)
+       ;; at <point>
+       ((equal prev "at") :transform)
+       ;; default
+       (t :none)))))
+
+(defvar renpy--label-definition-re
+  (renpy-rx label-keyword (1+ space) (group label-name))
+  "Regexp for looking up label definitions.")
+
+(defun renpy--collect-labels ()
+  "Return all unique label names in the current buffer."
+  (let (labels)
+    (save-excursion
+      (goto-char (point-min))
+      (while (re-search-forward renpy--label-definition-re nil t)
+        (push (match-string-no-properties 1) labels)))
+    labels))
+
+(defvar renpy--image-definition-re
+  (renpy-rx image-keyword (1+ space)
+	    (group image-name		; tag
+		   (0+ space name))	; attributes
+	    (0+ space) "=")
+  "Regexp for looking up image definitions.")
```

Comment:
Did you check every form of the commands where you are providing completion? As an example, the regex pattern to match an image misses images which are defined like this:
```renpy
image blackscreen:
    "black"
```

Reply:
good point, i'll have to extend this.